### ISSUE TITLE
Parsing Error (IMAP)View Log [home.pl compatibility]

### ISSUE THREAD

**Issue by MrTomek** [Foundry376/Mailspring#967]
Hello, I have a problem when trying to connect to home.pl servers

Parsing Error (IMAP):  
> ----------IMAP----------
> * OK projektyngo.home.pl IdeaImapServer 0.83 ready
> 1 CAPABILITY
> * CAPABILITY IMAP4rev1 LITERAL+ CHILDREN I18NLEVEL=1 IDLE SORT SORT=DISPLAY UIDPLUS UNSELECT QUOTA MOVE WITHIN LIST-EXTENDED LIST-STATUS SPECIAL-USE XLIST STARTTLS AUTH=PLAIN AUTH=LOGIN
> 1 OK Completed
> 2 LOGIN "t*****i@1ins.pl" "*********"
> * CAPABILITY IMAP4rev1 LITERAL+ CHILDREN I18NLEVEL=1 IDLE SORT SORT=DISPLAY UIDPLUS UNSELECT QUOTA MOVE WITHIN LIST-EXTENDED LIST-STATUS SPECIAL-USE XLIST STARTTLS
> 2 OK Completed
> 3 LIST "" ""
> * LIST (\Noselect) "." ""
> 3 OK Completed
> 4 XLIST "" "*"
> * XLIST (\HasNoChildren \Draft) "." "DRAFTS"
> * XLIST (\HasChildren \Inbox) "." "INBOX"
> * XLIST (\HasNoChildren \) "." "INBOX.ARCH"
> * XLIST (\HasNoChildren \Sent) "." "SENT"
> * XLIST (\HasNoChildren \Spam) "." "SPAM"
> * XLIST (\HasNoChildren \Trash) "." "TRASH"
> * XLIST (\HasNoChildren \) "." "confirmed-ham"
> * XLIST (\HasNoChildren \) "." "confirmed-spam"
> 4 OK Completed
> XLIST (\HasNoChildren \) "." "INBOX.ARCH"

**Comment by paranoiq1** [comment]
Did you manage to solve this? I have same problems on home.pl server. 
My log:
`----------IMAP----------
* OK serwer1479711.home.pl IdeaImapServer 0.83 ready
1 CAPABILITY
* CAPABILITY IMAP4rev1 LITERAL+ CHILDREN I18NLEVEL=1 IDLE SORT SORT=DISPLAY UIDPLUS UNSELECT QUOTA MOVE WITHIN LIST-EXTENDED LIST-STATUS SPECIAL-USE XLIST AUTH=PLAIN AUTH=LOGIN
1 OK Completed
2 LOGIN "marcin.tabakiernik" *********
* CAPABILITY IMAP4rev1 LITERAL+ CHILDREN I18NLEVEL=1 IDLE SORT SORT=DISPLAY UIDPLUS UNSELECT QUOTA MOVE WITHIN LIST-EXTENDED LIST-STATUS SPECIAL-USE XLIST
2 OK Completed
3 LIST "" ""
* LIST (\Noselect) "." ""
3 OK Completed
4 XLIST "" "*"
* XLIST (\HasNoChildren \Draft) "." "DRAFTS"
* XLIST (\HasNoChildren \) "." "Drafts"
* XLIST (\HasNoChildren \Inbox) "." "INBOX"
* XLIST (\HasNoChildren \Sent) "." "SENT"
* XLIST (\HasNoChildren \Spam) "." "SPAM"
* XLIST (\HasNoChildren \Trash) "." "TRASH"
4 OK Completed
XLIST (\HasNoChildren \) "." "Drafts"`

**Comment by pklebba** [comment]
Same problem with home.pl server too.

**Comment by kasia-basia** [comment]
Me too! Same problem.

**Comment by axxox** [comment]
Same problem with home.pl

**Comment by kilmariusz** [comment]
I have the same problem with home.pl

**Comment by xwojnar** [comment]
+1

**Comment by Krzemo94** [comment]
Hello,
Anyone solve it ?

**Comment by dtokarczyk** [comment]
I have the same problem with home.pl

**Comment by kwiatekx** [comment]
I have the same problem with home.pl / post.pl. Anyone have any solution?

**Comment by luski** [comment]
+1

**Comment by bprus** [comment]
+1

**Comment by bprus** [comment]
+1

**Comment by tomekd** [comment]
+1

 It seems there is a problem with the Polish popular mail provider `home.pl`.

**Comment by pakaado** [comment]
Any solved?

**Comment by RafalKrych** [comment]
I've got exactly the same problem. Please fix it....

**Comment by MrTomek** [comment]
@Foundry376 Fix this problem? Other email clients do not have this problem, but yours is the most proper

**Comment by bengotow** [comment]
Hey folks - thanks for reporting this. It looks like `home.pl` accounts aren't free and I'm a little confused which of their options you're using that is causing the problem. (It seems like one of them uses Google Apps Suite?)

If anyone has a home.pl domain and could temporarily create an email account I could test with, I should be able to get this fixed (or determine if it's a more serious problem.) I'd really appreciate it! Feel free to email me at ben@foundry376.com. Thanks 🙏

**Comment by RafalKrych** [comment]
Hello,
I've asked for help regarding this issue on home.pl helpdesk.
Hopefully someone from company will contact you and try to help with this.

**Comment by dtokarczyk** [comment]
@bengotow - I sent for you e-mail with account for tests. Good luck and we're waiting for fix :)

**Comment by dtokarczyk** [comment]
@RafalKrych Did you get info from Home.pl?

**Comment by bengotow** [comment]
Hey folks—sorry for the delay, this took a little while to track down, but I've got it fixed! Here's a quick rundown in case anyone stumbles across this in the future:

- Home.pl is using an IMAP server that calls itself "IdeaImapServer". I've never heard of this one before. It advertises the XLIST IMAP capability, but the XLIST response it sends cannot be parsed by Mailcore2. Mailspring now special cases "IdeaImapServer" and falls back to a more basic "LIST" request to get folder names.

- Home.pl doesn't provide message dates when you ask for "message headers" via IMAP. This is really strange - it's sort of one of the basic pieces of information along with message subjects. However, it does provide a "received date" (the date the message was received by the server rather than the date the sender pushed it through their SMTP server.) I've made Mailspring handle this case gracefully and I think this also fixes #457, which I've been hunting for for a LONG time.

- Home.pl's "Welcome to home.pl" email contained a bunch of inline attachments (nice little GIFs) but they didn't set a Content-Disposition indicating they were meant to be displayed in the message so they all showed up as attachments in Mailspring, which is crap. Mailspring now looks to see if the attachments are referenced in the HTML of the message, so we can detect that they're meant for inline display even if the MIME headers are missing. This should fix #219!

Thanks so much for your patience everyone—and @dtokarczyk for the test account! I have to do a bit of testing, but this will ship soon in the next Mailspring release.
